src/toml.rs

```diff
@@ -31,7 +66,45 @@ pub fn patch_manifest(
         source(working_dir, manifest_directory, mode),
     );
 
-    Ok(manifest.to_string())
+    Ok(())
+}
+
+fn remove_patch_from_manifest(
+    manifest_table: &mut toml_edit::Table,
+    name: &str,
+) -> anyhow::Result<()> {
+    if let Some(patch_table) = manifest_table.get_mut("patch") {
+        let mut to_remove_registry = None;
+
+        let patch_table = patch_table.as_table_mut().unwrap();
+        for (registry_name, patch_table_item) in patch_table.iter_mut() {
+            let registry_table = patch_table_item.as_table_mut().unwrap();
+            if registry_table.remove(name).is_some() {
+                if registry_table.is_empty() {
+                    to_remove_registry = Some(registry_name.to_owned());
+                }
+
+                // We can stop searching, it should be only one patch per package name.
+                break;
+            }
+        }
+
+        // TODO: somehow it removes the comment in the manifest file -> see test
+        //       Removes a comment in the final toml file when using the tool as well
+        //       Maybe it thinks the comment refers to the table.
+        //       Reason: sees a comment before a table as a decor which belongs to it
+        //       Solution: don't remove if there is any comment in front of the table?
+        //       Solution2: toml_edit should only take direct attached comments as prefix?
+        //       On the other hand it wouldn't be a problem if we add the patch section at the end,
+        //            there shouldn't be any comments before it, which do not belong to it.
+        //
```

Comment:
Is this comment still true? Looking at the tests, it looks to me like the comments are preserved.

I'm looking at:

https://github.com/eopb/cargo-override/blob/76132402d1afbd01cb0b8ed56894a70172f2a158/src/toml.rs#L279-L297

Reply:
First, when I tested, I put the patch section right after the comment `# See more ...`. This was removed, while removing the patch section. In this scenario, it would be a problem removing such a comment. There should be any other section when adding a patch section, so this scenario will never happen.

Do you have any thoughts on that? Otherwise, this can be kept in mind but will not lead to any problem as I see.

Reply:
Makes sense 👍

I'm okay with merging without fixing this one as long as we open an issue for it. After this is merged, and we have that issue, we can debug it separately and perhaps open a issue on `toml_edit` if it needs fixing there.

In the meantime, it would be good if you restored the test that you had that demonstrated the comment being removed. The test could be useful to reference in the issue since it would clearly show the undesirable behaviour.